stdlib.go

```diff
@@ -95,8 +110,10 @@ func (a StdlibAdapter) Write(p []byte) (int, error) {
 	if timestamp != "" {
 		keyvals = append(keyvals, a.timestampKey, timestamp)
 	}
-	if file, ok := result["file"]; ok && file != "" {
-		keyvals = append(keyvals, a.fileKey, file)
+	if a.expectCaller {
+		if file, ok := result["file"]; ok && file != "" {
```

Comment:
It's not clear that we need two `if` checks. If `a.expectCaller == false` then the regexp used in `subexps` should never put a "file" key in the map, so `ok` would always be `false` in that case, right?